### PR TITLE
Make operation timeout configurable - Closes #272

### LINKED ISSUE #272: Pod readiness checks should have configurable timeout values
In some environments (Travis CI, in my case), it can take longer than the hardcoded 60 seconds for the zookeeper and/or kafka pods to become available. This may be due to network slowness in downloading the container images or just general slowness of the cluster.

It would be great if these values were configurable in the Kubernetes/OpenShift yml definitions.

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- ~~Bugfix~~
- Enhancement / new feature
- ~~Refactoring~~

### Description

This PR makes it possible to configure the Operation timeout. This option is currently used only in KafkaClusteroperation where we use the `readiness` method. But we should over time use it also in other clusters. When that is the case, it should move to `AbstractClusterOperation` . But right now this was not needed.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java [hunk 1]
```diff
@@ -20,24 +20,29 @@
     public static final String STRIMZI_NAMESPACE = "STRIMZI_NAMESPACE";
     public static final String STRIMZI_CONFIGMAP_LABELS = "STRIMZI_CONFIGMAP_LABELS";
     public static final String STRIMZI_FULL_RECONCILIATION_INTERVAL = "STRIMZI_FULL_RECONCILIATION_INTERVAL";
+    public static final String STRIMZI_OPERATION_TIMEOUT = "STRIMZI_OPERATION_TIMEOUT";
 
     public static final long DEFAULT_FULL_RECONCILIATION_INTERVAL = 120_000; // in ms (2 minutes)
+    public static final long DEFAULT_OPERATION_TIMEOUT = 60_000; // in ms (1 minute)
```
Comment (tombentley) [inline cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java]: I like the Kafka habit of including the unit in the same: `DEFAULT_OPERATION_TIMEOUT_MS` would makes the unit unambiguous at the use site, and there's then no need for the comment. This can be deferred to another PR if you prefer.

File: cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java [hunk 2]
```diff
@@ -20,24 +20,29 @@
...
 
     private Map<String, String> labels;
     private Set<String> namespaces;
     private long reconciliationInterval;
+    private long operationTimeout;
```
Comment (tombentley) [inline cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java]: Likewise: `operationTimeoutMs`

File: cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java [hunk 3]
```diff
@@ -69,13 +74,19 @@ public static ClusterControllerConfig fromMap(Map<String, String> map) {
...
         }
-        Long reconciliationInterval = DEFAULT_FULL_RECONCILIATION_INTERVAL;
 
+        Long reconciliationInterval = DEFAULT_FULL_RECONCILIATION_INTERVAL;
         String reconciliationIntervalEnvVar = map.get(ClusterControllerConfig.STRIMZI_FULL_RECONCILIATION_INTERVAL);
         if (reconciliationIntervalEnvVar != null) {
             reconciliationInterval = Long.valueOf(reconciliationIntervalEnvVar);
         }
 
+        Long operationTimeout = DEFAULT_OPERATION_TIMEOUT;
```
Comment (tombentley) [inline cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterControllerConfig.java]: Is it really necessary for this to be a `Long`? You could use a primitive and `Long.parseLong()` instead of `Long.valueOf()`.

File: cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperations.java [hunk 4]
```diff
@@ -42,6 +42,8 @@
     private static final String CLUSTER_TYPE_KAFKA = "kafka";
     private static final String CLUSTER_TYPE_TOPIC_CONTROLLER = "topic-controller";
 
+    private final long operationTimeout;
```
Comment (tombentley) [inline cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperations.java]: Same comment: `operationTimeoutMs`

File: documentation/adoc/cluster-controller.adoc [hunk 5]
```diff
@@ -337,3 +337,6 @@ The controller itself can be configured through the following environment variab
 Default: `strimzi.io/kind=cluster`.  
 * `STRIMZI_FULL_RECONCILIATION_INTERVAL` : the interval between periodic reconciliations.
 Default: 120000 ms
+* `STRIMZI_OPERATION_TIMEOUT` : Timeout for internal operations specified in milliseconds. This value should be
```
Comment (tombentley) [inline documentation/adoc/cluster-controller.adoc]: Should we make clear that this timeout applies separately to each operation (SS readiness, Pod readiness, Service readiness) and not to the whole logical operation that's made up of those individual operations?

**Comment by scholzj** [comment]
@tombentley I think the `Ms` comments are valid, but they should be used also to the outside. I changed it for both user facing values in CC. I think we should changed it also for TC (perhaps in separate PR). WDYT?

File: documentation/adoc/cluster-controller.adoc [hunk 6]
```diff
@@ -246,13 +246,13 @@ The configurable fields are the following :
...
 Default is the namespace where the topic controller is running
-* `reconciliationInterval`: the interval between periodic reconciliations. Default is "15 minutes"
+* `reconciliationIntervalMs`: the interval between periodic reconciliations. Default is "15 minutes"
 * `zookeeperSessionTimeout`: the Zookeeper session timeout. Default is "20 seconds"
 
 .Example Topic Controller JSON configuration
 [source,json]
 ----
-{ "reconciliationInterval": "10 minutes", "zookeeperSessionTimeout": "10 seconds" }
+{ "reconciliationIntervalMs": "10 minutes", "zookeeperSessionTimeout": "10 seconds" }
```
Comment (tombentley) [inline documentation/adoc/cluster-controller.adoc]: This is currently erroneous (it's the TC parameter)

**Comment by tombentley** [comment]
@scholzj yes, I agree we should be consistent for the TC.

File: documentation/adoc/cluster-controller.adoc (hunk 5 above)
Comment (scholzj) [reply documentation/adoc/cluster-controller.adoc]: I don't think we should get into details. Or at least I'm not sure how to write it without spending too much time on describing hot the Cluster Controller works. I think the current level is sufficient.

**Comment by scholzj** [comment]
@tombentley The TC seems to accept different values in configuration which include the time unit. I guess I will raise a separate issue for that.
